Meadowlark-js/backends/meadowlark-postgresql-backend/src/repository/SqlHelper.ts

```diff
@@ -8,11 +8,16 @@ import format from 'pg-format';
 
 /**
  * Function that returns a SQL query for checking whether references exist for a given list of document ids
+ * This has a secondary function of locking this row for proper updating and deleting
  * @param documentId the document to check for existence
  * @returns SQL query to check for existence ids
  */
 export function existenceIdsForDocument(documentId: string): string {
-  return format(`SELECT existence_id FROM meadowlark.existence WHERE document_id = %L`, documentId);
+  return format(`SELECT existence_id FROM meadowlark.existence WHERE document_id = %L FOR SHARE NOWAIT`, documentId); //
+}
+
+export function existenceCheckForDocument(documentId: string): string {
```

Comment:
Can we make this more action-oriented? Maybe something like

```suggestion
export function lookForExistingDocument(documentId: string): string {
```